libs/application/core/src/types/ApplicationTemplate.ts

```diff
@@ -25,5 +26,8 @@ export interface ApplicationTemplate<
     states: StatesConfig<TContext, TStateSchema, TEvents> // TODO Extend StatesConfig to completely enforce meta being required attribute
   }
   readonly stateMachineOptions?: Partial<MachineOptions<TContext, TEvents>>
-  mapUserToRole(id: string, state: string): ApplicationRole
+  mapUserToRole(
+    id: string,
+    application: Application,
+  ): ApplicationRole | undefined
```

Comment:
Wondering about the return type here. If a user has no role, shouldn't we throw an error?

Reply:
i was wondering the same. Not sure what should happen. If there is no role, there will be no form loaded, nor any answers readable or writable any ways. I am a bit afraid of throwing an error, but maybe we can do that later ?